src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -641,35 +641,41 @@ public ICompilationTracker FreezePartialState(CancellationToken cancellationToke
                 if (state is FinalCompilationTrackerState { IsFrozen: true } finalState)
                     return this;
 
-                var projectState = state switch
-                {
-                    // If we don't have an existing state, then transition to a project state without any data.
-                    null => this.ProjectState.RemoveAllDocuments(),
+                var initialProjectState = GetInitialProjectState(state);
+                var (frozenProjectState, frozenState) = GetFrozenCompilationState(initialProjectState, state);
 
-                    FinalCompilationTrackerState => this.ProjectState,
+                Contract.ThrowIfFalse(frozenState.IsFrozen);
+                return new CompilationTracker(frozenProjectState, frozenState, this.SkeletonReferenceCache.Clone());
 
-                    // If we have an in progress state with no steps, then we're just at the current project state.
-                    InProgressState { PendingTranslationActions: [] } => this.ProjectState,
+                ProjectState GetInitialProjectState(CompilationTrackerState? state)
+                {
+                    return state switch
+                    {
+                        // If we don't have an existing state, then transition to a project state without any data.
+                        null => this.ProjectState,
```

Comment:
intentionally changes.  teh RemoveAllDocuments is now in GetFrozenCompilationState

Reply:
If that's the case then update the comment.

Reply:
And at this point, I just want to see GetIniitalProjectState combined with the other function. There's now two functions that both have the same switches and you can only understand how this works if I look at both of them at once. It's far more cognitive load to understand the structure than just one bigger method here.